lib/chef/provider/service/macosx.rb

```diff
@@ -77,6 +95,11 @@ def define_resource_requirements
               @current_resource.running(false)
             end
           end
+
+          requirements.assert(:all_actions) do |a|
+            a.assertion { @plist.nil? ? true : ::File.exists?(@plist) }
+            a.failure_message Chef::Exceptions::Service, "#{@plist} does not exist"
+          end
```

Comment:
oh i see now... Well this error seems like it might be descriptive enough. Should I just remove the assert? Or do what i was talking about with adding the assertion to  the 'find_service_label' function?

```
    ================================================================================
    Error executing action `enable` on resource 'macosx_service[com.CPE.daily]'
    ================================================================================

    Mixlib::ShellOut::ShellCommandFailed
    ------------------------------------
    Expected process to exit with [0], but received '1'
    ---- Begin output of plutil -convert xml1 -o - /Library/LaunchDaemons/com.CPE.daily.plist ----
    STDOUT: /Library/LaunchDaemons/com.CPE.daily.plist: file does not exist or is not readable or is not a regular file (Error Domain=NSCocoaErrorDomain Code=260 "The file “com.CPE.daily.plist” couldn’t be opened because there is no such file." UserInfo=0x7ff720414020 {NSFilePath=/Library/LaunchDaemons/com.CPE.daily.plist, NSUnderlyingError=0x7ff720413d90 "The operation couldn’t be completed. No such file or directory"})
    STDERR:
    ---- End output of plutil -convert xml1 -o - /Library/LaunchDaemons/com.CPE.daily.plist ----
    Ran plutil -convert xml1 -o - /Library/LaunchDaemons/com.CPE.daily.plist returned 1

    Cookbook Trace:
    ---------------
    /etc/chef/local-mode-cache/cache/cookbooks/cpe_utils/libraries/macosx.rb:220:in `find_service_label'
    /etc/chef/local-mode-cache/cache/cookbooks/cpe_utils/libraries/macosx.rb:51:in `load_current_resource'

    Resource Declaration:
    ---------------------
    # In /etc/chef/local-mode-cache/cache/cookbooks/cpe_ldi/recipes/default.rb

     19:   service daemon do
     20:     plist "#{ld_path}#{daemon}.plist"
     21:     action :enable
     22:   end
     23: end

    Compiled Resource:
    ------------------
    # Declared in /etc/chef/local-mode-cache/cache/cookbooks/cpe_ldi/recipes/default.rb:19:in `block in from_file'

    macosx_service("com.CPE.daily") do
      action [:enable]
      supports {:restart=>false, :reload=>false, :status=>false}
      retries 0
      retry_delay 2
      default_guard_interpreter :default
      service_name "com.CPE.daily"
      pattern "com.CPE.daily"
      plist "/Library/LaunchDaemons/com.CPE.daily.plist"
      declared_type :service
      cookbook_name "cpe_ldi"
      recipe_name "default"
    end
```